lib/entropy/dep.py

```diff
@@ -632,6 +632,21 @@ def is_valid_package_tag(tag):
         return False
     return True
 
+def _nat_sort_key(key):
+    """
+    Key generator for natural sorting algorithms.
+
+    @param key: Entropy package tag
+    return: array obtained from the input string with strings parsed as
+        strings and numbers parsed as number with the empty string as
+        first and last element.
+    """
+    def convert(text):
+        return int(text) if text.isdigit() else text.lower()
+    if key is None:
+        key = ''
+    return [convert(c) for c in re.split('([0-9]+)', key.strip())]
```

Comment:
can you compile this regexp and use the compiled version instead?

I am concerned about performance, as this is a function that is going to be called potentially a lot.
AFAIR:
I would create the variable at the module scope
SOMETING_REGEX = re.compile("([0-9]+)")

and then use SOMETHING_REGEX in the key function.

Reply:
Fixed in 5122204